tests/ttnn/unit_tests/operations/test_moreh_nll_loss_unreduced.py

```diff
@@ -0,0 +1,144 @@
+    npu_index_dtype = ttnn.int32
+
+    tt_target = to_npu(torch_target, device, npu_dtype=npu_index_dtype)
+    tt_weight = to_npu(torch_weight, device)
+    tt_output_grad = to_npu(torch_output_grad, device)
+    tt_input_grad = to_npu(torch_input_grad, device)
+
+    return tt_target, tt_weight, tt_output_grad, tt_input_grad
+
+
+def run_moreh_nll_loss_unreduced_backward(shape, ignore_index, none_weight, device, compute_kernel_options=None):
+    compute_kernel_config = get_compute_kernel_options(compute_kernel_options)
+
+    # run torch
+    (torch_input, torch_target, torch_weight, _) = get_torch_tensors(shape)
+    if none_weight:
+        torch_weight = None
+
+    if ignore_index == None:
+        nll_loss = torch.nn.NLLLoss(weight=torch_weight, reduction="none")
+    else:
+        nll_loss = torch.nn.NLLLoss(weight=torch_weight, ignore_index=ignore_index, reduction="none")
+    torch_loss = nll_loss(torch_input, torch_target)
+
+    output_grad = torch.randn_like(torch_loss)
+    torch_loss.backward(output_grad)
+
+    # run tt
+    (tt_target, tt_weight, tt_output_grad, tt_input_grad) = get_tt_backward_tensors(
+        torch_target, torch_weight, output_grad, torch_input.grad, device
+    )
+
+    tt_input_grad = ttnn.moreh_nll_loss_unreduced_backward(
+        tt_target,
+        tt_output_grad,
+        weight_tensor=tt_weight,
+        input_grad_tensor=tt_input_grad,
+        ignore_index=ignore_index,
+        compute_kernel_config=compute_kernel_config,
+    )
+    tt_input_grad_to_cpu = to_cpu(tt_input_grad, torch_input.grad.shape)
+
+    rtol = atol = 0.05
+    passing, out = comp_allclose_and_pcc(torch_input.grad, tt_input_grad_to_cpu, pcc=0.999, rtol=rtol, atol=atol)
+
+    # print("Output grad: ", output_grad)
+    # print("Weight: ", torch_weight)
+    # print("CPU: ", torch_input.grad)
+    # print("NPU: ", tt_input_grad_to_cpu)
+
+    logger.debug(f"Out passing (param)={passing}")
+    logger.debug(f"Output pcc={out}")
+
+    assert passing
+
+
+@pytest.mark.parametrize(
+    "shape",
+    [
+        # (32, 32),
+        # (400, 300),
+        # (20, 300, 320),
+        # (5, 2, 5, 40, 70),
+    ],
+)
+@pytest.mark.parametrize("ignore_index", [1])
+@pytest.mark.parametrize("none_weight", [True, False])
+@pytest.mark.parametrize("compute_kernel_options", compute_kernel_options, ids=compute_kernel_ids)
+def test_moreh_nll_loss_unreduced_backward(
+    shape, ignore_index, none_weight, compute_kernel_options, device, use_program_cache
+):
+    torch.manual_seed(0)
+
+    run_moreh_nll_loss_unreduced_backward(
+        shape, ignore_index, none_weight, device, compute_kernel_options=compute_kernel_options
+    )
+
+
+@pytest.mark.parametrize(
+    "shape",
+    [
+        (2, 3),
+        (2, 3, 4),
+        (2, 3, 5, 4),
+    ],
+)
+@pytest.mark.parametrize("none_weight", [True, False])
+@pytest.mark.parametrize("ignore_index", [0, None])
```

Comment:
https://pytorch.org/docs/stable/generated/torch.nn.functional.nll_loss.html#torch.nn.functional.nll_loss

According to the PyTorch specifications, `ignore_index` is of type int, so it does not support None type input.

Reply:
Actually, in the test, I use an if statement to check if `ignore_index` is None, pass nothing to to the loss function
<img width="797" alt="image" src="https://github.com/user-attachments/assets/3cc8223a-a54b-435a-b291-3675adb55f0b">

But this is quite us confusing so I updated to what described in this [cmt](https://github.com/tenstorrent/tt-metal/pull/12494#discussion_r1756075567).